### PR TITLE
Simplify aggregation hash table with code generation.

### PR DESCRIPTION
**Pull request by cpwright**

### CODE REVIEW

**Review by cpwright — COMMENTED** [review]
Placeholder.

**Review by cpwright — COMMENTED** [review]
Placeholder after rollup fix.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/ChunkedOperatorAggregationHelper.java [hunk 1]
```diff
@@ -179,7 +197,6 @@ private static QueryTable aggregation(AggregationControl control, SwapListener s
         ac.propagateInitialStateToOperators(result);
 
         if (withView.isRefreshing()) {
-            assert stateManager instanceof IncrementalChunkedOperatorAggregationStateManager;
```
[the remark this answers is not in the document]
Comment (cpwright) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/by/ChunkedOperatorAggregationHelper.java:182]: We determined not necessary on the phone.

**Review by cpwright — COMMENTED** [review]
Charles placeholder.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/OperatorAggregationStateManagerTypedBase.java [hunk 2]
```diff
@@ -55,119 +47,98 @@
...
-
-        ensureCapacity(tableSize);
     }
 
-    private void ensureCapacity(int tableSize) {
-        stateSource.ensureCapacity(tableSize);
-        overflowLocationSource.ensureCapacity(tableSize);
-        for (int ii = 0; ii < keySources.length; ++ii) {
-            keySources[ii].ensureCapacity(tableSize);
+    protected void ensureCapacity(int tableSize) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/OperatorAggregationStateManagerTypedBase.java:95]: `mainEnsureCapacity`? Consider following the same pattern as overflow, for symmetry, or annotating with `javax.annotation.OverridingMethodsMustInvokeSuper`.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+        return make(hasherConfigForBase(baseClass), tableKeySources, tableSize, maximumLoadFactor, targetLoadFactor);
+    }
+
+    @NotNull
+    public static <T> HasherConfig<T> hasherConfigForBase(Class<T> baseClass) {
+        final String packageMiddle;
+        final String mainStateName;
+        final String overflowStateName;
+        final String emptyStateName;
+        if (baseClass.equals(StaticChunkedOperatorAggregationStateManagerTypedBase.class)) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java:60]: Maybe we should have a classname prefix or suffix for this stuff, too.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException
+                | NoSuchMethodException e) {
+            throw new UncheckedDeephavenException("Could not instantiate " + castedClass.getCanonicalName(), e);
+        }
+        return retVal;
+    }
+
+    @NotNull
+    public static String packageName(String packageMiddle) {
+        return "io.deephaven.engine.table.impl.by.typed." + packageMiddle + ".gen";
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java:158]: You'll need more structure for joins.

Comment (cpwright) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java:165]: I will burn that bridge when/if we decide to get there.

**Pull request merged**